src/main/java/com/redi/demo/config/SecurityConfig.java

```diff
@@ -0,0 +1,44 @@
+package com.redi.demo.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(final HttpSecurity http) throws Exception {
+        http
+                .csrf().disable()
+                .authorizeRequests()
+                .antMatchers("/login").permitAll() /// everybody can login
+                .antMatchers(HttpMethod.POST, "/users").permitAll() // everybody can create a user
+                .antMatchers("/users/**").authenticated() // authenticated users
+                .antMatchers("/links").authenticated() // authenticated users
+                .antMatchers(HttpMethod.GET, "/**").permitAll() // everybody can expand the link
```

Comment:
This is not just links but rather everything. Perhaps use regext ant matching `/{w+}`